### PR TITLE
US_Census_PEP_Race

### PR DESCRIPTION
**Pull request by HarishC727**
Pull Request for Population Estimate By Race taken from US Census for the years 1900-2020

### CODE REVIEW

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 1]
```diff
@@ -0,0 +1,936 @@
...
+    df = df.groupby(['NAME', 'RACE']).sum().transpose().stack(0).reset_index()
+    if "2020" in file:
+        df[0] = df[1] + df[2] + df[3] + df[4] + df[5] + df[6]
+    df['0'] = df[0]
+    df['1'] = df[1]
+    df['2'] = df[2]
+    df['3'] = df[3]
+    df['4'] = df[4]
+    df['5'] = df[5]
+    df['6'] = df[6]
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:84]: Can column names be defined here itslef.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,936 @@
...
+
+    Arguments:
+        df (DataFrame) : DataFrame of xls dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for xls dataset.
+    """
+    df['Total People']=df[1]+df[2]+df[3]+df[4]+df[5]+df[6]+df[7]\
+        +df[8]+df[9]+df[10]+df[11]+df[12]+df[13]+df[14]+df[15]+\
+        df[16]+df[17]+df[18]
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:156]: for i in range(1,18):
    df['Total People'] += df[i]

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 3]
```diff
@@ -0,0 +1,936 @@
...
+    Arguments:
+        df (DataFrame) : DataFrame of xls dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for xls dataset.
+    """
+    # All the Column values are summed up to derive Total Population Value
+    df['Total People']=df[1]+df[2]+df[3]+df[4]+df[5]+df[6]+df[7]+\
+        df[8]+df[9]+df[10]+df[11]+df[12]+df[13]+df[14]+df[15]+\
+        df[16]+df[17]+df[18]
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:186]: Metion in comments what does these numbers mean

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 4]
```diff
@@ -0,0 +1,936 @@
...
+        df['30 to 34 years']+df['35 to 39 years'] +df['40 to 44 years']	\
+        +df['45 to 49 years'] +df['50 to 54 years'] +df['55 to 59 years']\
+        +df['60 to 64 years'] +df['65 to 69 years'] +df['70 to 74 years']\
+        +df['75 to 79 years'] +df['80 to 84 years'] +df['85 years and over']
+    df = df.drop(['Under 5 years','5 to 9 years','10 to 14 years',\
+        '15 to 19 years','20 to 24 years','25 to 29 years',\
+        '30 to 34 years','35 to 39 years','40 to 44 years',\
+        '45 to 49 years','50 to 54 years','55 to 59 years',\
+        '60 to 64 years','65 to 69 years','70 to 74 years','75 to 79 years',\
+        '80 to 84 years','85 years and over'],axis=1)
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:255]: Use list and loop - for adding.
list - for dropping.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 5]
```diff
@@ -0,0 +1,936 @@
...
+    if len(col) >= 15:
+        df['Black or African American Alone'] = pd.to_numeric(df['7'])
+        df = df.drop(["10", "11", "12"], axis=1)
+    else:
+        df['Non White'] = df['7']
+    df = df.drop(["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"], axis=1)
+    return df
+
+
+def _clean_county_20_csv_file(file_path: str) -> pd.DataFrame:
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:302]: Please add in doc string - which files this function is processing.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 6]
```diff
@@ -0,0 +1,936 @@
...
+    Also, Performs transformations on the data.
+
+    Arguments:
+        df (DataFrame) : DataFrame of csv dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for csv dataset.
+    """
+    df.drop(df.index[65:], inplace=True)
+    df.drop(df.index[1:14], inplace=True)
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:455]: Mention what columns are getting dopped here.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 7]
```diff
@@ -0,0 +1,936 @@
...
+        df (DataFrame) : Transformed DataFrame for txt dataset.
+    """
+    df['1'] = df['1'].astype(str)
+    mask = df['1'].str.len() >= 6
+    df = df.loc[mask]
+    mask = df['2'] >= 50000
+    df = df.loc[mask]
+    mask = df['1'].str[0] == '7'
+    df = df.loc[mask]
+    df = df[df['1'].str.contains("999")]
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:498]: Please add functional comment for this block

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 8]
```diff
@@ -0,0 +1,936 @@
...
+    else:
+        suffix = ".mcf"
+    return suffix
+
+
+def _mcf_process(col: str):
+    """
+    This method processes MCF file w.r.t
+        dataframe headers.
+    """
```
Comment (AravindhVenkatesh) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:634]: Change Doc String to Google format

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 5 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:302]: Mentioned the files being cleaned.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 1 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:84]: Renamed it.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 9]
```diff
@@ -0,0 +1,936 @@
...
+        """
+        tmcf_template = """Node: E:USA_Population_Count->E{}
+typeOf: dcs:StatVarObservation
+variableMeasured: dcs:{}
+measurementMethod: dcs:{}
+observationAbout: C:USA_Population_Count->geo_ID
+observationDate: C:USA_Population_Count->Year
+observationPeriod: \"P1Y\"
+value: C:USA_Population_Count->{}  
+"""
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:887]: indented the specified block

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 10]
```diff
@@ -0,0 +1,936 @@
...
+        Returns:
+            None
+        """
+        mcf_template = """Node: dcid:{}
+typeOf: dcs:StatisticalVariable
+populationType: dcs:Person
+statType: dcs:measuredValue
+measuredProperty: dcs:count
+race: dcs:{}
+"""
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:850]: indented the specified block

File: scripts/us_census/pep/population_estimate_by_race/preprocess_test.py [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+"""
+
+import os
+import unittest
+from os import path
+from preprocess import CensusUSAPopulationByRace
+# module_dir_ is the path to where this test is running from.
+module_dir_ = os.path.dirname(__file__)
+test_data_folder = os.path.join(module_dir_, "test_data")
+op_data_folder = os.path.join(module_dir_, "test_output_data")
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess_test.py:25]: converted variables into constants

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 12]
```diff
@@ -0,0 +1,936 @@
...
+default_input_path = os.path.dirname(
+    os.path.abspath(__file__)) + os.sep + "input_data"
+
+if ("ip_path" in sys.argv and len(sys.argv) > 2):
+    default_input_path = sys.argv[2]
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
+
+
+# Generating geoID by taking Geographical area as input
+def add_geo_id(df: pd.DataFrame) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:42]: made the function protected .

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 13]
```diff
@@ -0,0 +1,936 @@
...
+    def process(self):
+        """
+        This is main method to iterate on each file,
+        calls defined methods to clean, generate final
+        cleaned CSV file, MCF file and TMCF file.
+        """
+        for file in self.input_files:
+            df = self._load_data(file)
+            self._transform_data(df)
+        flag = 0
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:822]: replaced flag with direct suffix details and removed the function _mcf_path.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 14]
```diff
@@ -0,0 +1,936 @@
...
+            mcf = mcf + mcf_template.format(col, race) + "\n"
+
+        # Writing Genereated MCF to local path.
+        suffix = _mcf_path(flag)
+        with open(self.mcf_file_path+suffix,\
+                'w+', encoding='utf-8') as f_out:
+            f_out.write(mcf.rstrip('\n'))
+
+    # Generating TMCF file.
+    def _generate_tmcf(self, df_cols: list, flag: int) -> None:
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:868]: changed flag to measure and added a do string.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 3 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:186]: Added a comment mentioning what the number mean.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 15]
```diff
@@ -0,0 +1,936 @@
...
+
+pd.set_option("display.max_columns", None)
+
+_FLAGS = flags.FLAGS
+default_input_path = os.path.dirname(
+    os.path.abspath(__file__)) + os.sep + "input_data"
+
+if ("ip_path" in sys.argv and len(sys.argv) > 2):
+    default_input_path = sys.argv[2]
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:38]: changed it to default value.

File: scripts/us_census/pep/population_estimate_by_race/download1.sh [hunk 16]
```diff
@@ -0,0 +1,8 @@
+urls=$(<file_url.txt)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/download1.sh:1]: placed all the urls in one single file - url.json and having download.py according to it.
Deleted the file download1.sh

File: scripts/us_census/pep/population_estimate_by_race/download1.sh [hunk 17]
```diff
@@ -0,0 +1,8 @@
...
+
+mkdir -p "input_data"
+cd "input_data"
+for url in $urls;do
+    echo "downloading $url"
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/download1.sh:6]: deleted the file download1.sh as all the input files are being downloaded through download.py file

[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/download1.sh:6]: deleted the download1.sh file as all the input files are being downloaded through download.py file

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 2 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:156]: created loop for this above statement

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 4 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:255]: used loop to add the ages and dropped the list

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 7 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:498]: added comment

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 8 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:634]: changed it to doc string

**Review by Spaceenter — CHANGES_REQUESTED** [review]
Thanks for the great work! Left some style comments.

File: scripts/us_census/pep/population_estimate_by_race/download.py [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+                             on_bad_lines='skip',
+                             encoding='ISO-8859-1',
+                             low_memory=False)
+            df.to_csv(download_local_path + os.sep + file_name, index=False)
+        elif "CC-EST2020-ALLDATA" in _url:
+            df = pd.read_csv(_url,
+                             on_bad_lines='skip',
+                             encoding='ISO-8859-1',
+                             low_memory=False)
+            df.to_csv(download_local_path + os.sep + file_name, index=False)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/download.py:119]: Made it into one elif block.

File: scripts/us_census/pep/population_estimate_by_race/download.py [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+
+    elif ".txt" in _url:
+        if "srh" in _url:
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/download.py:122]: combined in one elif block.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 20]
```diff
@@ -0,0 +1,986 @@
...
+    df['White Alone'] = df[1]
+    df['Black or African American Alone'] = df[2]
+    df['American Indian or Alaska Native Alone'] = df[3]
+    df['Asian Alone'] = df[4]
+    df['Native Hawaiian and Other Pacific Islander Alone'] = df[5]
+    df['Two or more Races'] = df[6]
+    df['Year'] = df['level_0'].str[-4:]
+    df = df.drop(['level_0', 0, 1, 2, 3, 4, 5, 6], axis=1)
+    df.columns = df.columns.str.replace('NAME', 'Geographic Area')
+    # print(df.columns)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:90]: removed line.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 21]
```diff
@@ -0,0 +1,986 @@
...
+
+    Arguments:
+        df (DataFrame) : DataFrame of xls dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for xls dataset.
+     According to dataset, origin=0, sex=0
+     considered as they hold the total value.
+    """
+    #index=0
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:67]: removed the line.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 22]
```diff
@@ -0,0 +1,986 @@
...
+    Arguments:
+        df (DataFrame) : DataFrame of xls dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for xls dataset.
+    """
+    df = df.drop(['Census', 'Estimates Base'], axis=1)
+    df = df.drop([1], axis=0)
+    df.drop(df.index[7:], inplace=True)
+    #print(df.columns)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:108]: removed the line.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 23]
```diff
@@ -0,0 +1,986 @@
...
+    df['Unnamed: 0'] = df['Unnamed: 0'].str.replace(".", "")
+    df['Geographic Area'] = 'United States'
+
+    # it groups the df as per columns provided
+    # performs the provided functions on the data
+
+
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:115]: removed extra lines.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 24]
```diff
@@ -0,0 +1,986 @@
...
+    df.columns = df.columns.str.replace(
+        'Native Hawaiian and Other Pacific Islander'\
+        , 'Native Hawaiian and Other Pacific Islander Alone')
+    df.columns = df.columns.str.replace(
+        'American Indian and Alaska Native'\
+        , 'American Indian or Alaska Native Alone')
+    df.columns = df.columns.str.replace('Asian', 'Asian Alone')
+
+    df['Total'] = pd.to_numeric(df['Total'])
+    # print(df)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:132]: removed the line.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 25]
```diff
@@ -0,0 +1,986 @@
...
+    final_df = pd.DataFrame()
+    final_df = pd.concat([final_df, df])
+    df['geo_ID'] = df['geo_ID'].str[:-3]
+    df = df.groupby(['Year','geo_ID']).sum().\
+        reset_index()
+    df.drop(columns=['Total'], inplace=True)
+    # df.columns = df.columns.str.replace('White Alone',
+    #                                     'Count_Person_WhiteAlone')
+    # df.columns = df.columns.str.replace('Black or African American Alone',\
+    #         'Count_Person_BlackOrAfricanAmericanAlone')
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:211]: removed the extra lines.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 26]
```diff
@@ -0,0 +1,986 @@
...
+        '15': '70To74Years',
+        '16': '75To79Years',
+        '17': '80To84Years',
+        '18': '85OrMoreYears'
+    }
+    df = df.replace({"AGEGRP": conversion_of_agebracket_to_value})
+    # drop unwanted columns
+    df.drop(columns=['SUMLEV', 'STATE', 'COUNTY', 'STNAME', 'CTYNAME'], \
+        inplace=True)
+    # print(df)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:415]: removed the line.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 27]
```diff
@@ -0,0 +1,986 @@
...
+    df.columns = df.columns.str.replace('Area', 'Geographic Area')
+    df.columns = df.columns.str.replace('1', 'Total')
+    df.columns = df.columns.str.replace('2', 'Total White')
+    df.columns = df.columns.str.replace('5', 'Total Black')
+    df.columns = df.columns.str.replace('6', \
+        'Total American Indian & Alaska Native')
+    df.columns = df.columns.str.replace('7', 'Total Asian & Pacific Islander')
+
+    df["Geographic Area"] = [x.title() for x in df["Geographic Area"]]
+    #print(df)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:486]: removed all the print comands.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 28]
```diff
@@ -0,0 +1,986 @@
...
+        cleaned CSV file, MCF file and TMCF file.
+        """
+        for file in self.input_files:
+            print(file)
+            df = self._load_data(file)
+            self._transform_data(df)
+        # print(df)
+        name = "USA_Population_Count_by_Race_before_2000"
+        generator_df=pd.read_csv\
+            (os.path.join(self.cleaned_csv_file_path,"USA_Population_Count_by_Race_before_2000.csv"))
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:861]: Changes applied.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 29]
```diff
@@ -0,0 +1,986 @@
...
+        self._generate_mcf(generator_df.columns, name)
+        self._generate_tmcf(generator_df.columns, name)
+
+        name = "USA_Population_Count_by_Race_county_after_2000"
+        generator_df=pd.read_csv\
+            (os.path.join(self.cleaned_csv_file_path,"USA_Population_Count_by_Race_county_after_2000.csv"))
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:867]: Changes applied.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 30]
```diff
@@ -0,0 +1,986 @@
...
+        self._generate_mcf(generator_df.columns, name)
+        self._generate_tmcf(generator_df.columns, name)
+
+        name = "USA_Population_Count_by_Race_National_state_2000"
+        generator_df=pd.read_csv\
+            (os.path.join(self.cleaned_csv_file_path,"USA_Population_Count_by_Race_National_state_2000.csv"))
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:873]: Changes applied.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 31]
```diff
@@ -0,0 +1,986 @@
...
+            race = ""
+            if col.lower() in [
+                    "geographic area", "year", "short_form", "geo_id"
+            ]:
+                continue
+            race = _mcf_process(col)
+            mcf = mcf + mcf_template.format(col, race) + "\n"
+
+        # Writing Genereated MCF to local path.
+        # suffix = _mcf_path(flag)
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:908]: Changes applied.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 32]
```diff
@@ -0,0 +1,986 @@
...
+            "Count_Person_AsianOrPacificIslander",\
+            "Count_Person_TwoOrMoreRaces","Count_Person_NonWhite"]]
+        df_before_2000 = self.df[self.df["Year"] < 2000]
+        df_county_after_2000 = self.df[(self.df["Year"] >= 2000) &
+                                       (self.df["geo_ID"] != "country/USA") &
+                                       (self.df["geo_ID"].str.len() > 9)]
+        df_national_state_2000 = self.df[(self.df["Year"] >= 2000) &
+                                         ((self.df["geo_ID"].str.len() <= 9) |
+                                          (self.df["geo_ID"] == "country/USA"))]
+        # &(self.df["geo_ID"]=="country/USA")
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:834]: Changes applied.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 33]
```diff
@@ -0,0 +1,986 @@
...
+    Arguments:
+        df (DataFrame) : DataFrame of xls dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for xls dataset.
+     According to dataset, origin=0, sex=0
+     considered as they hold the total value.
+    """
+    #index=0
+    if "2020" in file:
```
[the remark this answers is not in the document]
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:68]: Added a comment.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 34]
```diff
@@ -0,0 +1,971 @@
...
+""" A Script to process USA Census PEP monthly population data
+    from the datasets in provided local path.
+    Typical usage:
+    1. python3 preprocess.py
+    2. python3 preprocess.py -i input_data
+"""
+import os
+import re
+import sys
+# For import util.alpha2_to_dcid
```
Comment (Spaceenter) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:23]: Pls remove unused code.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py [hunk 35]
```diff
@@ -0,0 +1,971 @@
...
+    os.path.join(os.path.dirname(__file__), '../../../../'))
+sys.path.insert(1, _COMMON_PATH)
+
+import pandas as pd
+from absl import app
+from absl import flags
+from util.alpha2_to_dcid import USSTATE_MAP
+from states_to_shortform import get_states
+
+# pd.set_option("display.max_columns", None)
```
Comment (Spaceenter) [inline scripts/us_census/pep/population_estimate_by_race/preprocess.py:34]: Pls remove unused code.

**Review by Spaceenter — APPROVED** [review]

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 35 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:34]: Removed the comment.

File: scripts/us_census/pep/population_estimate_by_race/preprocess.py (hunk 34 above)
Comment (HarishC727) [reply scripts/us_census/pep/population_estimate_by_race/preprocess.py:23]: This is a comment stating that alpha2_to_dcid is being imported from util for geoId. I made the comment more specific.

**Pull request merged**